CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFile.swift

```diff
@@ -54,7 +55,17 @@ final class CEWorkspaceFile: Codable, Comparable, Hashable, Identifiable, Editor
     /// If the item already is the top-level ``CEWorkspaceFile`` this returns `nil`.
     var parent: CEWorkspaceFile?
 
-    var fileDocument: CodeFileDocument?
+    var fileDocument: CodeFileDocument? {
+        didSet {
+            fileDocumentSubject.send()
+        }
+    }
+
+    /// Publisher for fileDocument property
+    var fileDocumentPublisher: AnyPublisher<Void, Never> {
+        fileDocumentSubject.eraseToAnyPublisher()
+    }
+    private let fileDocumentSubject = PassthroughSubject<Void, Never>()
```

Comment:
Maybe define the variable above its usages

Reply:
Done